### ISSUE TITLE
[macOS] Drop .pkg in favor of .dmg/.app distribution

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#5926]
While including "private links" feature (https://github.com/owncloud/client/pull/5763) in 2.4 we decided to drop support for macOS versions older than 10.9 to avoid maintaining two implementations for the same problem (already solved in newer APIs with the Finder Extensions). This allows us to simplify packaging for the client on macOS. 

@jturcotte what would be the preferred way to distribute the client once we get rid of the `.pkg`s? 

We'll also need to update the docs regarding OS support policy: https://doc.owncloud.org/server/latest/admin_manual/installation/system_requirements.html#desktop once 2.4 is planned (we could do that in master already cc/ @settermjd )

**Comment by jturcotte** [comment]
I think beside .pkg, the only way I know is a .dmg like you mentioned. There are still issues where we run a script to enable our FinderSync extension, and I think we won't be able to run this command with a .dmg. This could be problematic unless we ask the user to enable/disable the extension in the client's configuration instead of system configuration (like Dropbox is possibly still doing).

**Comment by guruz** [comment]
Is there a reason to switch?
The pkg works (after a lot of effort spent with it).

I actually like pkg better because I can just run it, I don't need to do the awkward mount-then-move-the-app-to-some-finder-window-then-unmount

**Comment by SamuAlfageme** [comment]
> I actually like pkg better because I can just run it, I don't need to do the awkward mount-then-move-the-app-to-some-finder-window-then-unmount 

Kind of agree, yeah. I opened this one 'cause I remember we talked about this when the private links were about to happen and I thought there could be some side benefit from it (maybe smaller installers (current pkgs are 27MB), not entering the password when installing or avoiding the install wizard for a simpler, drag-and-drop procedure). 

@jturcotte I think Dropbox distributes a separate, lightweight `.app` (container dmg is ~700KB) that runs installation tasks - not the client itself - and it somehow manages to auto-enable the Finder Extension on System Preferences:

![extensions](https://user-images.githubusercontent.com/2644445/28707237-eacdf0d2-7377-11e7-952e-851447fd9205.png)

Close here?
